chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -0,0 +1,113 @@
+#
+# Cookbook Name:: crowbar
+# Recipe:: crowbar-upgrade
+#
+# Copyright 2013-2016, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This recipe is for executing actions that need to be done at nodes
+# as a preparation for the system upgrade. The preparation itself
+# consists of several steps which we distinguish by various attributes
+# saved in the node structure.
+
+return unless node[:platform] == "suse"
+
+if node["crowbar_wall"]["crowbar_openstack_upgrade"]
+
+  # Actions to be run last, when admin node is already new SUSE Cloud version (6)
+  # Nodes will be restarted and their system upgraded after this.
+
+  # put HA node into maintenance mode (TODO is it needed? It wasn't in 4-5 upgrade...)
+
+  # Stop pacemaker resources in all clusters
+  bash "stop pacemaker resources" do
+    code <<-EOF
+      for type in clone ms primitive; do
+        for resource in $(crm configure show | grep ^$type | cut -d " " -f2); do
+          crm resource stop $resource
+        done
+      done
+    EOF
+    only_if { ::File.exist?("/usr/sbin/crm") }
+  end
+
+  # Stop openstack services
+  bash "stop HA and openstack services" do
```

Comment:
Hm, I'm not sure... first we do not know which services are running there, so we are looking for them with that pattern... that way it would be ugly to use first bash and than go back to chef resource.

But maybe... we actually have the node data, so maybe we could look inside and actually know which services should be running? But i'm not sure if we have such mapping somewhere in the node or we would need manually created mapping (like `:keystone => [ "openstack-keystone"] ` etc.)